### PR TITLE
Fixed index ordering to not have shuffled output

### LINKED ISSUE #1981: idxs_inv are currently broken so that output datasets are shuffled
### What happened?

idxs_inv are currently broken so that output datasets are shuffled

### What are the steps to reproduce the bug?

_No response_

### Hedgedoc link to logs and more information. This ticket is public, do not attach files directly.

_No response_

### PR DESCRIPTION
**Pull request by clessig**
## Description

Fixed index ordering to not have shuffled output

## Issue Number

Closes #1981 

## Checklist before asking for review

-   [ ] I have performed a self-review of my code
-   [ ] My changes comply with basic sanity checks:
      - I have fixed formatting issues with `./scripts/actions.sh lint`
      - I have run unit tests with `./scripts/actions.sh unit-test`
      - I have documented my code and I have updated the docstrings.
      - I have added unit tests, if relevant
-   [ ] I have tried my changes with data and code:
      - I have run the integration tests with `./scripts/actions.sh integration-test`
      - (bigger changes) I have run a full training and I have written in the comment the run_id(s): `launch-slurm.py --time 60`
      - (bigger changes and experiments) I have shared a hegdedoc in the github issue with all the configurations and runs for this experiments
-   [ ] I have informed and aligned with people impacted by my change:
    - for config changes: the MatterMost channels and/or a design doc
    - for changes of dependencies: the MatterMost software development channel

### CODE REVIEW

**Comment by clessig** [comment]
coords going into validation_io:

tensor([[   2.3377, -135.0000],
        [   1.4026, -135.4545],
        [   1.4026, -134.5455],
        [   0.4675, -135.0000],
        [   3.2727, -133.6082],
        [   2.3377, -134.0816],
        [   2.3377, -133.1633],
        [   1.4026, -133.6364],
        [   3.2727, -136.3918],
        [   2.3377, -136.8367],
        [   2.3377, -135.9184],
        [   1.4026, -136.3636],
        [   4.2078, -135.0000],
        [   3.2727, -135.4639],
        [   3.2727, -134.5361],
        [   4.2078, -132.1875],
        [   3.2727, -132.6804],
        [   3.2727, -131.7526],
        [   5.1428, -131.6842],
        [   5.1428, -130.7368],
        [   4.2078, -131.2500],
        [   4.2078, -130.3125],
        [   5.1428, -134.5263],
        [   5.1428, -133.5789],
        [   5.1428, -132.6316],
        [   4.2078, -134.0625],
        [   4.2078, -133.1250],
        [   6.0779, -133.0851],
        [   6.0779, -132.1277],
        [   6.0779, -131.1702],
        [   4.2078, -137.8125],
        [   3.2727, -138.2474],
        [   3.2727, -137.3196],
        [   5.1428, -137.3684],
        [   5.1428, -136.4211],
        [   5.1428, -135.4737],
        [   4.2078, -136.8750],
        [   4.2078, -135.9375],
        [   5.1428, -139.2632],
        [   5.1428, -138.3158],
        [   4.2078, -139.6875],
        [   4.2078, -138.7500],
        [   6.0779, -138.8298],
        [   6.0779, -137.8723],
        [   6.0779, -136.9149],
        [   6.0779, -135.9574],
        [   6.0779, -135.0000],
        [   6.0779, -134.0426],
        [   7.9480, -134.0217],
        [   7.0130, -134.5161],
        [   7.0130, -133.5484],
        [   7.0130, -132.5806],
        [   7.9480, -135.9783],
        [   7.0130, -137.4194],
        [   7.0130, -136.4516],
        [   7.0130, -135.4839],
        [   8.8831, -135.4945],
        [   8.8831, -134.5055],
        [   7.9480, -135.0000],
        [   6.0779, -130.2128],
        [   6.0779, -129.2553],
        [   6.0779, -128.2979],
        [   5.1428, -129.7895],
        [   7.9480, -128.1522],
        [   7.0130, -128.7097],
        [   7.0130, -127.7419],
        [   7.0130, -126.7742],
        [   7.9480, -131.0870],
        [   7.0130, -131.6129],
        [   7.0130, -130.6452],
        [   7.0130, -129.6774],
        [   8.8831, -129.5604],
        [   8.8831, -128.5714],
        [   7.9480, -130.1087],
        [   7.9480, -129.1304],
        [   8.8831, -126.5934],
        [   7.9480, -127.1739],
        [   7.9480, -126.1957],
        [   9.8181, -126.0000],
        [   9.8181, -125.0000],
        [   8.8831, -125.6044],
        [   8.8831, -124.6154],
        [   9.8181, -129.0000],
        [   9.8181, -128.0000],
        [   9.8181, -127.0000],
        [   8.8831, -127.5824],
        [  11.6883, -126.8182],
        [  10.7532, -127.4157],
        [  10.7532, -126.4045],
        [  10.7532, -125.3933],
        [   8.8831, -132.5275],
        [   8.8831, -131.5385],
        [   7.9480, -133.0435],
        [   7.9480, -132.0652],
        [   9.8181, -131.0000],
        [   9.8181, -130.0000],
        [   8.8831, -130.5495],
        [   9.8181, -134.0000],
        [   9.8181, -133.0000],
        [   8.8831, -133.5165]])
After reordering:

>>> t_coords[:100]
tensor([[  89.2842,    0.0000],
        [  89.2842,   18.0000],
        [  89.2842,   36.0000],
        [  89.2842,   54.0000],
        [  89.2842,   72.0000],
        [  89.2842,   90.0000],
        [  89.2842,  108.0000],
        [  89.2842,  126.0000],
        [  89.2842,  144.0000],
        [  89.2842,  162.0000],
        [  89.2842, -180.0000],
        [  89.2842, -162.0000],
        [  89.2842, -144.0000],
        [  89.2842, -126.0000],
        [  89.2842, -108.0000],
        [  89.2842,  -90.0000],
        [  89.2842,  -72.0000],
        [  89.2842,  -54.0000],
        [  89.2842,  -36.0000],
        [  89.2842,  -18.0000],
        [  88.3570,    0.0000],
        [  88.3570,   15.0000],
        [  88.3570,   30.0000],
        [  88.3570,   45.0000],
        [  88.3570,   60.0000],
        [  88.3570,   75.0000],
        [  88.3570,   90.0000],
        [  88.3570,  105.0000],
        [  88.3570,  120.0000],
        [  88.3570,  135.0000],
        [  88.3570,  150.0000],
        [  88.3570,  165.0000],
        [  88.3570, -180.0000],
        [  88.3570, -165.0000],
        [  88.3570, -150.0000],
        [  88.3570, -135.0000],
        [  88.3570, -120.0000],
        [  88.3570, -105.0000],
        [  88.3570,  -90.0000],
        [  88.3570,  -75.0000],
        [  88.3570,  -60.0000],
        [  88.3570,  -45.0000],
        [  88.3570,  -30.0000],
        [  88.3570,  -15.0000],
        [  87.4243,    0.0000],
        [  87.4243,   12.8571],
        [  87.4243,   25.7143],
        [  87.4243,   38.5714],
        [  87.4243,   51.4286],
        [  87.4243,   64.2857],
        [  87.4243,   77.1429],
        [  87.4243,   90.0000],
        [  87.4243,  102.8571],
        [  87.4243,  115.7143],
        [  87.4243,  128.5714],
        [  87.4243,  141.4286],
        [  87.4243,  154.2857],
        [  87.4243,  167.1429],
        [  87.4243, -180.0000],
        [  87.4243, -167.1429],
        [  87.4243, -154.2857],
        [  87.4243, -141.4286],
        [  87.4243, -128.5714],
        [  87.4243, -115.7143],
        [  87.4243, -102.8571],
        [  87.4243,  -90.0000],
        [  87.4243,  -77.1429],
        [  87.4243,  -64.2857],
        [  87.4243,  -51.4286],
        [  87.4243,  -38.5714],
        [  87.4243,  -25.7143],
        [  87.4243,  -12.8571],
        [  86.4904,    0.0000],
        [  86.4904,   11.2500],
        [  86.4904,   22.5000],
        [  86.4904,   33.7500],
        [  86.4904,   45.0000],
        [  86.4904,   56.2500],
        [  86.4904,   67.5000],
        [  86.4904,   78.7500],
        [  86.4904,   90.0000],
        [  86.4904,  101.2500],
        [  86.4904,  112.5000],
        [  86.4904,  123.7500],
        [  86.4904,  135.0000],
        [  86.4904,  146.2500],
        [  86.4904,  157.5000],
        [  86.4904,  168.7500],
        [  86.4904, -180.0000],
        [  86.4904, -168.7500],
        [  86.4904, -157.5000],
        [  86.4904, -146.2500],
        [  86.4904, -135.0000],
        [  86.4904, -123.7500],
        [  86.4904, -112.5000],
        [  86.4904, -101.2500],
        [  86.4904,  -90.0000],
        [  86.4904,  -78.7500],
        [  86.4904,  -67.5000],
        [  86.4904,  -56.2500]])

File: src/weathergen/datasets/tokenizer_masking.py [hunk 1]
```diff
@@ -213,19 +213,10 @@ def get_target_values(
...
-        #     device_sel = selection.to(data.device)
-        #     data = data.index_select(0, device_sel)
-        #     coords = coords.index_select(0, device_sel)
-        #     if idxs_ord_inv.numel() > 0:
-        #         idxs_ord_inv = idxs_ord_inv.index_select(0, device_sel)
+        if data.numel() > 0:
+            # flatten per-token indices into one flat list
+            idxs_flat = torch.cat([idxs for idxs_cell in idxs_cells for idxs in idxs_cell])
+            # compute indices for inversion
+            _, idxs_ord_inv = torch.sort(idxs_flat)
```
Comment (shmh40) [inline src/weathergen/datasets/tokenizer_masking.py:221]: Don't we need to handle the empty data, data.numel() = 0, case?

Comment (shmh40) [reply src/weathergen/datasets/tokenizer_masking.py:221]: Otherwise we have no idxs_ord_inv and hence this function will fail at the return

Comment (clessig) [reply src/weathergen/datasets/tokenizer_masking.py:221]: Yes!

**Review by shmh40 — APPROVED** [review]
